### ISSUE TITLE
Android Auto does not show current location unless OsmAnd is open on the phon

### ISSUE THREAD

**Issue by mdykierek** [osmandapp/OsmAnd#19839]
### Description

When setting route from the car's screen, the route and car position and direction does not show.
I have to open OsmAnd+ UI on the phone to show center the map and show vehicle position and heading.

### Steps to reproduce

Set a route on car's screen and drive.

### Actual result

Car's location and heading is not shown on the car's screen.

### Expected result

Location and route should be centered and shown without need to open OsmAnd on the phone.

### Your Environment (required)

**WARNING** Crash-Logs **MAY** contain information you deem sensitive.
Review this **CAREFULLY** before posting your issue!

```
OsmAnd Version:
Android/iOS version:
Device model:
Crash-Logs: ?
```

**Comment by yuriiurshuliak** [comment]
Please indicate the application version you are using.
What GPS permissions have been granted: Continuous or One-time use?
Which location source is employed: Google Play Services or Android API?
Have you enabled device/battery optimization?

**Comment by mdykierek** [comment]
1. OsmAnd+ version 4.7.17
2. Android 14 on. Motorola Edge 40 Pro
3. Android Auto 11.8.641624-release
4. Toyota Corlolla '21
5. GPS access is granted for "When using App" - there is no continuous option.
6. Location Source is Google Play Services
7. I usually do have battery saver enabled, but Android Auto requires USB connection and this means the device is being charged. When charging battery saver is irrelevant and marked as unavailable.

I will try to figure out more exact steps to reproduce as sometimes it occurs more often. It definitely is not a one time thing. I avoid reporting such things as I treat them as my mistakes or lack of understanding how thing work.

**Comment by mdykierek** [comment]
So I nailed it.
The behavior can be observed when the destination is on/behind a private access road. My house is located in such place.
So I leave the office, set destination=Home and drive. The voice prompts, the lane assistance work fine, but the map does not show up:
![aa](https://github.com/osmandapp/OsmAnd/assets/27214784/d0142677-1213-4296-a414-1eb2aaba8bf0)
So you need to unlock the phone an open the OsmAnd to see that is asks a question there:
![ss](https://github.com/osmandapp/OsmAnd/assets/27214784/18ae0eae-21ba-4e0d-adb3-7b750df47d21)
You don't need to answer. The map on car's screen is back to normal.
My workaround is to move the Home to a public access road and the problem is gone. But if for example you put in an address that is private, you are in trouble.

**Comment by DmitryAlexei** [comment]
I tried several times to reproduce your issue on release version (OsmAnd 4.7.17, released: 2024-05-07) and head unit simulator, but almost every time it worked fine. 
- I set my work favorite destination located in access=private zone
- I switched off osmand on the phone, phone screen was also locked
- allow private access option on the phone was disabled
- I entered work as my destination using android auto head unit and started navigation
Actual result
- despite warning on the phone (see screenshot) navigation instructions on the AA screen were correct 
- only one time location on the android auto was not synchronized with my current location. in that case, I could zoom out map manually and see my current location. Turn instructions were also correct

![telegram-cloud-photo-size-2-5391366739389961471-y](https://github.com/osmandapp/OsmAnd/assets/116079541/446301b7-016f-47ab-9713-0d0a01256abd)

**Comment by Galbar** [comment]
I'm having the same issue. I've also noticed that it gets solved if I open the app on the phone and hit button to center the map on my location. Tapping on the button on the Android Auto display does not work when this is happening.
The rest of the navigation works fine, though. It is just the map visualization.

**Comment by mdykierek** [comment]
Two pieces of news:
1. Tried private access. Results were strange. For example route to an airport first led through hotel alleys and used starting lanes and restricted areas.
2. The worse thing is, that today I had it twice and the destination nor any road on the way was private. So my "nailed" is for one of possible scenarios only.

**Comment by mdykierek** [comment]
Two videos that show the behavior. This week I have it all the time, every ride I make. Sorry for the quality, but they show our point. The second video shows that the GPS data is coming in and the speed widget updates correctly. The rest of display does not.

https://github.com/osmandapp/OsmAnd/assets/27214784/249dc2a4-6d79-4ebe-ba33-9d1be65f1451

https://github.com/osmandapp/OsmAnd/assets/27214784/1ad7e94d-4d8a-48d6-8ecb-c8943ecd9fe8

**Comment by mdykierek** [comment]
A new symptom. After the navigation has completed, the display froze. Even "Free ride" did not display the location and movement.

**Comment by SnejPro** [comment]
I observed the same issue. I think the reason is that I can not grant OsmAnd access to the location while in background.

![Screenshot_20240611-165544.png](https://github.com/osmandapp/OsmAnd/assets/11321022/78228ef8-1fd6-4a31-a4a4-aa76e225389c)



When I open the app it works fine.
I have the issue regardless whether the destination is in a private road.

Greetings 
jens

**Comment by SnejPro** [comment]
Device: Google Pixel 8a
Android: 14
Version: 4.7.17 (Google Play)

**Comment by mdykierek** [comment]
The location access permission is not the reason. Let me explain why I think so:
1. The app is on the car's screen, so most probably it is in use.
2. For me, having the app open and active causes slow refresh (another report), so I have to put it in background and the problem does not come back.
3. The navigation works, it knows the speed. For some reason map is "frozen". Maybe it's the on-screen menu/start navigation in Android Auto that "misbehave"? I don't know. It is hard to drive and "debug".

**Comment by SnejPro** [comment]
> The location access permission is not the reason. Let me explain why I think so:
> 1. The app is on the car's screen, so most probably it is in use.
> 2. For me, having the app open and active causes slow refresh (another report), so I have to put it in background and the problem does not come back.
> 3. The navigation works, it knows the speed. For some reason map is "frozen". Maybe it's the on-screen menu/start navigation in Android Auto that "misbehave"? I don't know. It is hard to drive and "debug".

Sound's plausible. But maybe it is worth a try  allowing the users to grant osmand "background" position permissions? It would be an easy fix.

**Comment by mdykierek** [comment]
I did report such an issue in the past. There was a version that would not work for me if OsmAnd was not open and in foreground.
The permanent setting supposedly is gone since Android 11: #13732.
In my case showing UI on the phone is enough to fix Android Auto. Then it can be put in background and the phone can be locked.

**Comment by Galbar** [comment]
I have noticed that it happens after I have used the map app outside of the car (i.e. to look for places while on foot). Maybe another hint?

**Comment by DmitryAlexei** [comment]
@Galbar @SnejPro @mdykierek Please check if OsmAnd 4.8 on Android Auto works correctly for you.

**Comment by mdykierek** [comment]
Well. It installed yesterday. Didn't work today, but the situation was different than before. I didn't have location on when OsmAnd opened on the AA, so I still need to confirm the "usual" scenario.
This no location situation leads me to a feature request - ALL other navigation apps I have display clear prompt asking to enable location. OsmAnd displays non functional map with some location I don't know.
Such a prompt screen is useful.

**Comment by Pick-Nick10** [comment]
I'm also experiencing this issue

OsmAnd Version: 4.8.4
Android/iOS version: osmand+ google play android version
Device model: pixel 7
Other things to note: I'm using BRouter for my routing

**Comment by mdykierek** [comment]
With 4.8.4 there is one nice thing. The navigation stays when AA is disconnected and then reconnected. This however leads to this issue - every time the route is restored, the location is gone.

**Comment by vshcherb** [comment]
Yes, we fixed navigation route and navigation service in 4.8.5 (beta).
However we still experience 2 issues:
1. Sometimes when AA disconnected / restarted, instead of correct navigation profile, Browse map is selected and straight line route is built
2. Indeed sometimes location is not updated, especially if app was never started/resumed on the phone (after force stop or termination), it's visible as buttons my location / Zoom in / zoom out don't work on the panel. We reproduced but not so often after fixes with new Target SDK of Android 14 and navigation services

it looks like on Android issue, we will try to workaround by addition all location provider listeners. As workaround you could try switch between: Android / Google Play location service provider in OsmAnd Settigns
